src/TimeModeling/Utils/auxiliaryFunctions.jl

```diff
@@ -444,7 +444,7 @@ function setup_3D_grid(xrec::Array{Array{T, 1}, 1},yrec::Array{Array{T, 1},1},zr
             for j=1:nxrec
                 xloc[i][idx] = xrec[i][j]
                 yloc[i][idx] = yrec[i][k]
-                zloc[i][idx] = zrec[i]
+                zloc[i][idx] = ( length(JUDI.tof32(zrec)) == 1 ? zrec[1] : zrec[i])
```

Comment:
No the case "==1" should not be an array and should be caught line 466 for a single zrec value. If there is an `if` the multiple dispatch isn't set correctly or some weird input is there.